sentinel/src/transitions/accept_patient_reports.js

```diff
@@ -114,6 +114,38 @@ const addRegistrationToDoc = (doc, registrations) => {
   }
 };
 
+const addReportUUIDToRegistration = (doc, registrations) => {
+  if (registrations.length) {
+    var visitReportedDate = doc.reported_date;
+
+    var messageFound = false;
+    registrations.forEach(function(registration) {
+      if (registration.scheduled_tasks) {
+        registration.scheduled_tasks.forEach(function(task, index) {
+          if (!messageFound) {
+            if (
+              (task.state === 'delivered' || task.state === 'sent') &&
+              task.messages
+            ) {
+              var nextTask = registration.scheduled_tasks[index + 1];
```

Comment:
What happens when the current task you're iterating over is the last one? Does that invalidate it for the report_uuid being attached to it?

I think this logic needs a comment to explain your reasoning.